# 01 Seasonality

Do WTI's calendar-month mean returns from before 2020 show up again after it?

Holdout starts 2020-01-01. Verdict: held.

This study reads stored series. It does not add a derived column.

In [1]:
import json
from pathlib import Path
studies = json.loads(Path('data/studies.json').read_text())
study = next(item for item in studies['studies'] if item['slug'] == 'seasonality')
print(study['verdict'])
print(study['verdict_line'])
print(study['table']['caption'])
for row in study['table']['rows']:
    print(' | '.join(row))


held
Correlation of the 12 train month-means with the 12 holdout month-means is 0.309. Signs match in 5 of 12 months. The correlation is above zero, which was the bar. 5 matching signs out of 12 is a thin map.
Mean daily log return of RWTC by calendar month
01 | 638 | -0.042% | 141 | 0.172% | not the same
02 | 603 | -0.032% | 134 | 0.081% | not the same
03 | 676 | 0.170% | 154 | -0.180% | not the same
04 | 625 | 0.192% | 143 | 0.448% | same
05 | 633 | -0.001% | 146 | 0.285% | not the same
06 | 655 | 0.030% | 146 | 0.041% | same
07 | 638 | 0.007% | 149 | 0.220% | same
08 | 707 | 0.059% | 153 | -0.167% | not the same
09 | 634 | 0.039% | 144 | -0.003% | not the same
10 | 677 | -0.106% | 129 | -0.038% | same
11 | 630 | -0.201% | 117 | -0.185% | same
12 | 651 | -0.001% | 128 | 0.140% | not the same


## Method

Train prints end 31 Dec 2019. Holdout prints start 1 Jan 2020. Each calendar month gets a mean daily log return in each window.

## Result

Correlation of the 12 train month-means with the 12 holdout month-means is 0.309. Signs match in 5 of 12 months. The correlation is above zero, which was the bar. 5 matching signs out of 12 is a thin map.